api/types/app.go

```diff
@@ -422,3 +423,89 @@ func (a Apps) Less(i, j int) bool { return a[i].GetName() < a[j].GetName() }
 
 // Swap swaps two apps.
 func (a Apps) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+const (
+	jwtClaimRewriteTypeNoneString           = "none"
+	jwtClaimRewriteTypeRolesString          = "roles"
+	jwtClaimRewriteTypeRolesAndTraitsString = "roles-and-traits"
+)
+
+func (r JWTClaimsRewriteType) encode() (string, error) {
+	switch r {
+	case JWTClaimsRewriteType_JWT_REWRITE_UNSPECIFIED:
+		return "", nil
+	case JWTClaimsRewriteType_JWT_REWRITE_ROLES_AND_TRAITS:
+		return jwtClaimRewriteTypeRolesAndTraitsString, nil
+	case JWTClaimsRewriteType_JWT_REWRITE_ROLES:
+		return jwtClaimRewriteTypeRolesString, nil
+	case JWTClaimsRewriteType_JWT_REWRITE_NONE:
+		return jwtClaimRewriteTypeNoneString, nil
+	}
+	return "", trace.BadParameter("invalid jwt rewrite type %v", r)
+}
+
+func (r *JWTClaimsRewriteType) decode(val any) error {
+	var valS string
+	switch val := val.(type) {
```

Comment:
You can just assert directly to string: `val.(string)` - no need for a switch if you're only ever handling one case.